modules/Core/pages/register.php

```diff
@@ -245,152 +210,123 @@
 
                 // Check if there was any integrations errors
                 if (!isset($integration_errors)) {
-                    // Minecraft user account association
-                    if (isset($account_verification) && $account_verification == '1') {
-                        // MCAssoc enabled
-                        // Get data from database
-                        $mcassoc_site_id = SITE_NAME;
+                    $user = new User();
 
-                        $mcassoc_shared_secret = DB::getInstance()->get('settings', ['name', 'mcassoc_key'])->results();
-                        $mcassoc_shared_secret = $mcassoc_shared_secret[0]->value;
-
-                        $mcassoc_instance_secret = DB::getInstance()->get('settings', ['name', 'mcassoc_instance'])->results();
-                        $mcassoc_instance_secret = $mcassoc_instance_secret[0]->value;
-
-                        define('MCASSOC', true);
+                    $ip = HttpUtils::getRemoteAddress();
+                    if (!filter_var($ip, FILTER_VALIDATE_IP)) {
+                        // TODO: Invalid IP, do something
+                    }
 
-                        // Hash password first
-                        $password = password_hash($_POST['password'], PASSWORD_BCRYPT, ['cost' => 13]);
-                        $_SESSION['password'] = $password;
-                        unset($_POST['password']);
+                    $password = password_hash(Input::get('password'), PASSWORD_BCRYPT, ['cost' => 13]);
+                    // Get current unix time
+                    $date = new DateTime();
+                    $date = $date->getTimestamp();
```

Comment:
```suggestion
                    $date = date('U');
```

Reply:
I guess you could technically just move this definition straight to the $user->create() part

Reply:
Done